lib/authz/permissions.go

```diff
@@ -336,6 +336,10 @@ var ErrIPPinningMissing = trace.AccessDenied("pinned IP is required for the user
 // ErrIPPinningMismatch is returned when user's pinned IP doesn't match observed IP.
 var ErrIPPinningMismatch = trace.AccessDenied("pinned IP doesn't match observed client IP")
 
+// ErrIPPinningNotAllowed is returned when user's pinned IP doesn't match observed IP.
+var ErrIPPinningNotAllowed = trace.AccessDenied("IP pinning is not allowed for connections behind L4 load balancers with " +
+	"PROXY protocol enabled without explicitly setting 'proxy_protocol: on' in the proxy_service or auth_service config.")
```

Comment:
Should it be `proxy_service and auth_service config`?

Reply:
Depends on the client setup, if there's another LB between Proxy and Auth then it can be both, although usually I think it will be only one. I'll make it `and/or`.